Sources/CacheAdvance/CacheReader.swift

```diff
@@ -64,6 +65,11 @@ final class CacheReader {
                 // If the previous read was also empty, then the file has been corrupted.
                 throw CacheAdvanceError.fileCorrupted
             }
+            if seekForwardOnly && startingOffset > FileHeader.expectedEndOfHeaderInFile {
+                // If seekForwardOnly is true, then return nil when startingOffset is ahead
+                // Otherwise, there will be an infinite loop in reading messages.
+                return nil
```

Comment:
since we're returning early here, can we use a `guard` rather than an `if`? That way it's more clear to future engineers that this block of code will always return